Bottom: a4f930386d1079ec83f29bd0c722114770f6554d
Top:    3913d172c2d623b9aa46a39a6b725092c2b0d3ca
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-21 16:46:34 -0700

Refresh of conn-err

---

diff --git a/examples/rstream.c b/examples/rstream.c
index cd8a962..34ff421 100644
--- a/examples/rstream.c
+++ b/examples/rstream.c
@@ -481,8 +481,11 @@ static int client_connect(void)
 
 		len = sizeof err;
 		ret = getsockopt(rs, SOL_SOCKET, SO_ERROR, &err, &len);
-		if (ret || err) {
-			ret = ret ? ret : err;
+		if (ret)
+			goto err;
+		if (err) {
+			ret = rdma_seterrno(err);
+			perror("async rconnect")
 			goto err;
 		}
 	}
